### DISCUSSION TITLE
Can't get `ignore-paths:` to do what I want

### DISCUSSION THREAD

**Discussion by deivid-rodriguez** [community/community#25957] [Actions]
<p>Hi!</p>
<p>I’m trying to get my workflow skipped if any of the changed paths by a push to the master branch is <code>bundler/CHANGELOG.md</code>. This is what I’m doing:</p>
<pre><code class="lang-yaml">on:
  push:
    branches:
      - master

    paths-ignore:
      - bundler/CHANGELOG.md
</code></pre>
<p>However, it doesn’t seem to be working. See <a href="https://github.com/rubygems/rubygems/actions/runs/169757578/workflow" rel="nofollow noopener">https://github.com/rubygems/rubygems/actions/runs/169757578/workflow</a>, where the workflow run even though the push was changing <code>bundler/CHANGELOG.md</code>.</p>
<p>I think I must be missing something very obvious but I can’t see it.</p>
<p>Thanks so much!</p>

**Comment by deivid-rodriguez** [comment]
<p>Oh! I think it might be due to the fact that the <code>paths-ignore</code> configuration was introduced in that same push? If this is it, it should no longer be a problem I guess. I’ll verify if the next time it doesn’t happen.</p>

**Comment by weide-zhou** [comment, marked as answer]
<p>Hi <a class="mention" href="/deivid-rodriguez">@deivid-rodriguez</a>,</p>
<p>If your commits contains other files which out of ‘path-ignore’ configurations, the push event will be triggered.</p>
<p>For your setting, if you only change in bundler/CHANGELOG.md in master branch, the workflow  will NOT be triggered.</p>
<p>As an alternative, you can get the file list from the ${{ github.sha }}(link <a href="https://github.community/t/check-pushed-file-changes-with-git-diff-tree-in-github-actions/17220/2">here</a>), check if the bundler/CHANGELOG.md exists, skip the jobs then.</p>
<p>Thanks.</p>

**Comment by deivid-rodriguez** [comment]
<p>Mmmmm, right.</p>
<p>Aren’t <a href="https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#example-ignoring-paths" rel="nofollow noopener">the docs</a> a bit confusing?</p>
<blockquote>
<p>Anytime a path name matches a pattern in <code>paths-ignore</code> , the workflow will not run.</p>
</blockquote>
<p>This sounds exactly like what I was interpreting.</p>
<p>Although the example later on actually explains the real behavior.</p>
<p>I’ll try an alternative, thanks!!</p>
